architecture/additional_concepts/dynamic_admission_controllers.adoc

```diff
@@ -0,0 +1,536 @@
+
+* xref:admission-webhooks-m[Mutating admission webhooks] allow for the use of mutating webhooks to modify resource content before it is persisted.
+
+* xref:admission-webhooks-v[Validating admission webhooks] allow for the use of validating webhooks to enforce custom admission policies.   
+
+Configuring the webhooks and external webhook servers is beyond the scope of this document. However, the webhooks must adhere to an
+https://github.com/kubernetes/kubernetes/blob/v1.10.0-beta.1/staging/src/k8s.io/api/admission/v1beta1/types.go#L28[interface]
+in order to work properly with {product-title}. 
+
+[IMPORTANT]
+====
+*Webhook admissions* are Technology Preview features only.
+ifdef::openshift-enterprise[]
+Technology Preview features are not supported with Red Hat production service
+level agreements (SLAs), might not be functionally complete, and Red Hat does
+not recommend to use them for production. These features provide early access to
+upcoming product features, enabling customers to test functionality and provide
+feedback during the development process.
+
+For more information on Red Hat Technology Preview features support scope, see
+https://access.redhat.com/support/offerings/techpreview/.
+endif::[]
+====
+
+When an object is instantiated, {product-title} makes an API call to admit the object. During the admission process, a _mutating admission controller_ 
+can invoke admission webhooks to perform tasks, such as injecting affinity labels. At the end of the admissions process, 
+a _validating admission controller_ can invoke admission webhooks to make sure the object is configured properly, such as verifying affinity labels. 
+If the validation passes, {product-title} schedules the object as configured.
+
+The following diagram illustrates this process with two admission webhooks that call multiple webhooks.
+
+image::api-server-pipeline.png["API admission stage", align="center"]
+
+A simple example use case for admission webhooks is syntactical validation
+of resources. For example, you have an infrastructure that requires all pods to
+have a common set of labels, and you do not want any pod to be
+persisted if the pod does not have those label. You could write a webhook to inject these labels 
+and another webhook to verify that the labels are present.
+The {product-title} will then schedule schedule that have the labels and pass validation
+and reject pods that do not pass due to missing labels.
+
+Some common use-cases include:
+
+* Mutating resources to inject side-car containers into pods. 
+* Restricting projects to block some resources from a project. 
+* Complex custom resource validation to perform complex validation on dependent fields.
+
+=== Types of Admission Webhooks
+
+Cluster administrators can include _mutating admission webhooks_ or _validating admission webhooks_ 
+in the admission chain of the API server.
+
+//http://blog.kubernetes.io/2018/01/extensible-admission-is-beta.html
```

Comment:
Why you comment out this instead of remove?

Reply:
@wanghaoran1988 I put this in so that I would remember where the information came from, should someone ask. I think i can remove at this point.